### DISCUSSION TITLE
add some nice features

### DISCUSSION THREAD

**Discussion by cowarol** [tobychui/zoraxy#397] [Ideas]
First of all thank you for your project.
Can you add some features on your TODO Please. Can you publish your todo?
- Docker labels
- internal LDAP/AD (all in one with SSO for other services like Proxmox/portainer/...)
- SSO (WIP)
- ssh console (like guacamole but only the ssh console)
- add tailscale/twingate tunnel
- crowdsec
- add cname record to the DNS provider via api - with a slider (extern/intern use) 
- monitor for other services (like uptime kuma)

**Comment by tobychui** [comment]
> Can you publish your todo?

Well I can't because I don't have one. After I am done with one feature, I just pick a random one from the Issue enhancement list that seems to be most wanted and can be implemented in my remaining spare time of the month. I probably should invite someone to help do some project management tasks for the Zoraxy project, but currently I am a bit too busy with the code base re-factorize after we got the first CVE. 

> Docker labels

No, docker support is not my primary goal for this project. Though, I always welcome PRs that improve  or optimize docker support and UX.

> internal LDAP/AD 

I did think about this, but many of my friends from the security field told me to not developing my own security related features that might cause even more maintenance workload. So this is not possible with our current limited man-hours from our contributors per month. 

> ssh console 

We already have that in the network tools tab

> add tailscale/twingate tunnel

I am planning to remove Zerotier from the core and move it to the WIP plugin system. So these are mostly likely not implemented by me but some other future plugin devs.

> crowdsec

Same as the reason for LDAP

> add cname record to the DNS provider via api - with a slider (extern/intern use)

Not sure what you mean by CNAME and DNS (Zoraxy doesn't have a DNS sever built in) . Zoraxy is developed in mind that we try to minimize external dependencies to 3rd party API. With ACME as an exception because there isn't really a better alternative. If what you are asking is some kind of private DNS server, no, currently we have no plan for this.

> monitor for other services

Might consider about this
